gk/fib.c

```diff
@@ -1254,9 +1259,17 @@ init_grantor_fib_locked(
 		gt_fib = &ltbl->fib_tbl6[fib_id];
 
 	gt_fib->action = GK_FWD_GRANTOR;
-	rte_memcpy(&gt_fib->u.grantor.gt_addr,
-		gt_addr, sizeof(gt_fib->u.grantor.gt_addr));
-	gt_fib->u.grantor.eth_cache = eth_cache;
+	gt_fib->u.grantor.set = rte_malloc("gk_fib.grantor.set",
+		sizeof(*(gt_fib->u.grantor.set)) +
+			sizeof(*(gt_fib->u.grantor.set->entries)), 0);
+	if (gt_fib->u.grantor.set == NULL) {
+		GK_LOG(ERR, "Could not allocate set of Grantor entries\n");
+		return NULL;
```

Comment:
Replace the `return NULL;`  with ` goto init_fib;`.

Reply:
Notice that this is already being corrected later in patch **fib: allow multiple Grantors per entry**, but it should be done in this patch instead.

Reply:
Also, don't forget to update the line `goto init_fib;` below, which is handling the failure of `lpm_add_route()`.

Reply:
Use `unlikely()` on the expression `gt_fib->u.grantor.set == NULL`.